packages/dd-trace/test/plugins/agent.js

```diff
@@ -35,13 +35,47 @@ function ciVisRequestHandler (request, response) {
   })
 }
 
+function handleTraceRequest (req, res, sendToTestAgent) {
+  // handles the received trace request and sends trace to Test Agent if bool enabled.
+  if (sendToTestAgent) {
+    const testAgentUrl = process.env.DD_TEST_AGENT_URL || 'http://127.0.0.1:9126'
+
+    // remove incorrect headers
+    delete req.headers['host']
+    delete req.headers['content-type']
+    delete req.headers['content-length']
+
+    const testAgentReq = http.request(
+      `${testAgentUrl}/v0.4/traces`, {
+        method: 'PUT',
+        headers: {
+          ...req.headers,
+          'X-Datadog-Agent-Proxy-Disabled': 'True',
+          'Content-Type': 'application/json'
+        }
+      })
+
+    testAgentReq.write(JSON.stringify(req.body))
+    testAgentReq.end()
+  }
+
+  res.status(200).send({ rate_by_service: { 'service:,env:': 1 } })
+  handlers.forEach(({ handler, spanResourceMatch }) => {
+    const trace = req.body
+    const spans = trace.flatMap(span => span)
+    if (isMatchingTrace(spans, spanResourceMatch)) {
+      handler(trace)
+    }
+  })
+}
+
 const DEFAULT_AVAILABLE_ENDPOINTS = ['/evp_proxy/v2']
 
 let availableEndpoints = DEFAULT_AVAILABLE_ENDPOINTS
 
 module.exports = {
   // Load the plugin on the tracer with an optional config and start a mock agent.
-  async load (pluginName, config, tracerConfig = {}) {
+  async load (pluginName, config, tracerConfig = {}, useTestAgent = true) {
```

Comment:
Any way to avoid having to change all callers? Why not just always try to call the test agent?

Reply:
I didn't have to change the majority of callers since this defaults to True and sends to Test Agent. For a few test suites, mainly the core tracer tests, I did change the call and set this variable to False so that we're not using the Test Agent, since we're mainly interested in traces produced by plugins.

Reply:
But can we avoid the caller even having to know about this? Why would it only sometimes be used? And why should the caller know about that?

Reply:
Thanks for the idea! I implemented a check before the mock agent loads to see if the test agent is running, then depending on if it is, sends/doesn't send traces. That removed the need to change this caller and simplified the PR a bunch